In [25]:
from scipy import stats
import pandas as pd
import numpy as np

In [27]:
ls

 Volume in drive C is Windows-SSD
 Volume Serial Number is 6271-5CB1

 Directory of C:\Users\gheva\Desktop\KOMAL Python

13-11-2024  15:45    <DIR>          .
20-10-2024  19:26    <DIR>          ..
10-11-2024  19:50    <DIR>          .ipynb_checkpoints
09-11-2024  21:14           103,118 Basic Stats 1 Assignment.ipynb
13-11-2024  15:45             3,978 Basic Stats2 Assignment.ipynb
               2 File(s)        107,096 bytes
               3 Dir(s)  49,292,726,272 bytes free


In [29]:
import os
os.getcwd()

'C:\\Users\\gheva\\Desktop\\KOMAL Python'

In [33]:
durability_df=pd.read_csv("C:/Users/gheva/OneDrive/Desktop/Durability data.csv")
durability_df

Sr.No.  Durability
0      1.0        1.13
1      2.0        1.55
2      3.0        1.43
3      4.0        0.92
4      5.0        1.25
5      6.0        1.36
6      7.0        1.32
7      8.0        0.85
8      9.0        1.07
9     10.0        1.48
10    11.0        1.20
11    12.0        1.33
12    13.0        1.18
13    14.0        1.22
14    15.0        1.29
15     NaN         NaN

##99% Confidence interval using sample standard deviation ##

For calculating the confidence inteval for this problem we use here t distribution because in the data sample standard deviation is provided instead of population standard devation.

In [55]:
durability_df.ci=stats.t.interval(0.99,14,loc=durability_df.Durability.mean(),scale=durability_df.Durability.std())
durability_df.ci

(0.6636474310163384, 1.813685902316995)

In [59]:
print('Mean at 99% Confidence interval:',np.round(durability_df.ci,2))

Mean at 99% Confidence interval: [0.66 1.81]


## Interpretation##
The mean durability of print-heads in terms of the number of characters printed before failure lies between 0.66 to 1.81

## 99% Confidence Interval using known population standard deviation##

In [63]:
durability_df.ci=stats.norm.interval(0.99,loc=durability_df.Durability.mean(),scale=durability_df.Durability.std())
durability_df.ci

(0.7411088413267759, 1.7362244920065573)

In [65]:
print('Mean at 99% Confidence interval:',np.round(durability_df.ci,2))

Mean at 99% Confidence interval: [0.74 1.74]


## Interpretation##
The mean durability of print-heads in terms of the number of characters printed before failure lies between 0.74 to 1.74